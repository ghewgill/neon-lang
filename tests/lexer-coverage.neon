-- Punctuation
( ) [ ] { } + * / ^ & = , . - < > :
− × ∕ ÷ ≠ ≤ ≥ ≔ ¬ ∧ ∨ ∈

-- Digraphs and trigraphs
-> <> <= >= := ...

-- Identifers
Aa Bb Cc Dd Ee Ff Gg Hh Ii Jj Kk Ll Mm Nn Oo Pp Qq Rr Ss Tt Uu Vv Ww Xx Yy Zz A_
aa bb cc dd ee ff gg hh ii jj kk ll mm nn oo pp qq rr ss tt uu vv ww xx yy zz a_
aA bB cC dD eE fF gG hH iI jJ kK lL mM nN oO pP qQ rR sS tT uU vV wW xX yY zZ
a1 b2 c3 d4 e5 f6 g7 h8 i9 j0 k1 l2 m3 n4 o5 p6 q7 r8 s9 t0 u1 v2 w3 x4 y5 z6

-- Keywords
IF THEN ELSE END WHILE DO VAR FUNCTION RETURN FALSE TRUE MOD AND OR NOT FOR TO
STEP Array Dictionary TYPE RECORD ENUM CONSTANT IMPORT IN OUT INOUT ELSIF CASE
WHEN EXIT NEXT LOOP REPEAT UNTIL DECLARE EXCEPTION TRY RAISE POINTER
NEW NIL VALID LET FIRST LAST AS DEFAULT EXPORT PRIVATE NATIVE FOREACH INDEX
ASSERT EMBED ALIAS IS BEGIN MAIN HEXBYTES INC DEC _ OTHERS WITH CHECK GIVES
NOWHERE INTDIV LABEL CLASS TRAP EXTENSION INTERFACE IMPLEMENTS UNUSED ISA
OPTIONAL IMPORTED TESTCASE EXPECT

-- Numbers
0b1101
0o755
0xbeef
1234
1234.5678
1234.5678e5
1234.5678e+5
1234.5678e-5
1_234.567_8e1_2

-- Strings
"abcd"
"a\"bcd"
"a\\bcd"
"a\bbcd"
"a\fbcd"
"a\nbcd"
"a\rbcd"
"a\tbcd"
"a\u007ebcd"
"a\U0000007ebcd"
"a\u{latin small letter a}"
"a\(x)bcd"
"a\(x:x)bcd"

-- Raw strings
@"abcd"
@@"abcd"@@
@tt@"abcd"@tt@

-- Comments
-- Single line comment
/* Multiple line
 * comment
 */
a/* */b
a/*
*/b
