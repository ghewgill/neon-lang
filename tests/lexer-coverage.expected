<2:1 LPAREN>
<2:3 RPAREN>
<2:5 LBRACKET>
<2:7 RBRACKET>
<2:9 LBRACE>
<2:11 RBRACE>
<2:13 PLUS>
<2:15 TIMES>
<2:17 DIVIDE>
<2:19 EXP>
<2:21 CONCAT>
<2:23 EQUAL>
<2:25 COMMA>
<2:27 DOT>
<2:29 MINUS>
<2:31 LESS>
<2:33 GREATER>
<2:35 COLON>
<3:1 MINUS>
<3:3 TIMES>
<3:5 DIVIDE>
<3:7 DIVIDE>
<3:9 NOTEQUAL>
<3:11 LESSEQ>
<3:13 GREATEREQ>
<3:15 ASSIGN>
<3:17 NOT>
<3:19 AND>
<3:21 OR>
<3:23 IN>
<6:1 ARROW>
<6:4 NOTEQUAL>
<6:7 LESSEQ>
<6:10 GREATEREQ>
<6:13 ASSIGN>
<6:16 ELLIPSIS>
<9:1 IDENTIFIER:Aa>
<9:4 IDENTIFIER:Bb>
<9:7 IDENTIFIER:Cc>
<9:10 IDENTIFIER:Dd>
<9:13 IDENTIFIER:Ee>
<9:16 IDENTIFIER:Ff>
<9:19 IDENTIFIER:Gg>
<9:22 IDENTIFIER:Hh>
<9:25 IDENTIFIER:Ii>
<9:28 IDENTIFIER:Jj>
<9:31 IDENTIFIER:Kk>
<9:34 IDENTIFIER:Ll>
<9:37 IDENTIFIER:Mm>
<9:40 IDENTIFIER:Nn>
<9:43 IDENTIFIER:Oo>
<9:46 IDENTIFIER:Pp>
<9:49 IDENTIFIER:Qq>
<9:52 IDENTIFIER:Rr>
<9:55 IDENTIFIER:Ss>
<9:58 IDENTIFIER:Tt>
<9:61 IDENTIFIER:Uu>
<9:64 IDENTIFIER:Vv>
<9:67 IDENTIFIER:Ww>
<9:70 IDENTIFIER:Xx>
<9:73 IDENTIFIER:Yy>
<9:76 IDENTIFIER:Zz>
<9:79 IDENTIFIER:A_>
<10:1 IDENTIFIER:aa>
<10:4 IDENTIFIER:bb>
<10:7 IDENTIFIER:cc>
<10:10 IDENTIFIER:dd>
<10:13 IDENTIFIER:ee>
<10:16 IDENTIFIER:ff>
<10:19 IDENTIFIER:gg>
<10:22 IDENTIFIER:hh>
<10:25 IDENTIFIER:ii>
<10:28 IDENTIFIER:jj>
<10:31 IDENTIFIER:kk>
<10:34 IDENTIFIER:ll>
<10:37 IDENTIFIER:mm>
<10:40 IDENTIFIER:nn>
<10:43 IDENTIFIER:oo>
<10:46 IDENTIFIER:pp>
<10:49 IDENTIFIER:qq>
<10:52 IDENTIFIER:rr>
<10:55 IDENTIFIER:ss>
<10:58 IDENTIFIER:tt>
<10:61 IDENTIFIER:uu>
<10:64 IDENTIFIER:vv>
<10:67 IDENTIFIER:ww>
<10:70 IDENTIFIER:xx>
<10:73 IDENTIFIER:yy>
<10:76 IDENTIFIER:zz>
<10:79 IDENTIFIER:a_>
<11:1 IDENTIFIER:aA>
<11:4 IDENTIFIER:bB>
<11:7 IDENTIFIER:cC>
<11:10 IDENTIFIER:dD>
<11:13 IDENTIFIER:eE>
<11:16 IDENTIFIER:fF>
<11:19 IDENTIFIER:gG>
<11:22 IDENTIFIER:hH>
<11:25 IDENTIFIER:iI>
<11:28 IDENTIFIER:jJ>
<11:31 IDENTIFIER:kK>
<11:34 IDENTIFIER:lL>
<11:37 IDENTIFIER:mM>
<11:40 IDENTIFIER:nN>
<11:43 IDENTIFIER:oO>
<11:46 IDENTIFIER:pP>
<11:49 IDENTIFIER:qQ>
<11:52 IDENTIFIER:rR>
<11:55 IDENTIFIER:sS>
<11:58 IDENTIFIER:tT>
<11:61 IDENTIFIER:uU>
<11:64 IDENTIFIER:vV>
<11:67 IDENTIFIER:wW>
<11:70 IDENTIFIER:xX>
<11:73 IDENTIFIER:yY>
<11:76 IDENTIFIER:zZ>
<12:1 IDENTIFIER:a1>
<12:4 IDENTIFIER:b2>
<12:7 IDENTIFIER:c3>
<12:10 IDENTIFIER:d4>
<12:13 IDENTIFIER:e5>
<12:16 IDENTIFIER:f6>
<12:19 IDENTIFIER:g7>
<12:22 IDENTIFIER:h8>
<12:25 IDENTIFIER:i9>
<12:28 IDENTIFIER:j0>
<12:31 IDENTIFIER:k1>
<12:34 IDENTIFIER:l2>
<12:37 IDENTIFIER:m3>
<12:40 IDENTIFIER:n4>
<12:43 IDENTIFIER:o5>
<12:46 IDENTIFIER:p6>
<12:49 IDENTIFIER:q7>
<12:52 IDENTIFIER:r8>
<12:55 IDENTIFIER:s9>
<12:58 IDENTIFIER:t0>
<12:61 IDENTIFIER:u1>
<12:64 IDENTIFIER:v2>
<12:67 IDENTIFIER:w3>
<12:70 IDENTIFIER:x4>
<12:73 IDENTIFIER:y5>
<12:76 IDENTIFIER:z6>
<15:1 IF>
<15:4 THEN>
<15:9 ELSE>
<15:14 END>
<15:18 WHILE>
<15:24 DO>
<15:27 VAR>
<15:31 FUNCTION>
<15:40 RETURN>
<15:47 FALSE>
<15:53 TRUE>
<15:58 MOD>
<15:62 AND>
<15:66 OR>
<15:69 NOT>
<15:73 FOR>
<15:77 TO>
<16:1 STEP>
<16:6 ARRAY>
<16:12 DICTIONARY>
<16:23 TYPE>
<16:28 RECORD>
<16:35 ENUM>
<16:40 CONSTANT>
<16:49 IMPORT>
<16:56 IN>
<16:59 OUT>
<16:63 INOUT>
<16:69 ELSIF>
<16:75 CASE>
<17:1 WHEN>
<17:6 EXIT>
<17:11 NEXT>
<17:16 LOOP>
<17:21 REPEAT>
<17:28 UNTIL>
<17:34 DECLARE>
<17:42 EXCEPTION>
<17:52 TRY>
<17:56 RAISE>
<17:62 POINTER>
<18:1 NEW>
<18:5 NIL>
<18:9 VALID>
<18:15 LET>
<18:19 FIRST>
<18:25 LAST>
<18:30 AS>
<18:33 DEFAULT>
<18:41 EXPORT>
<18:48 PRIVATE>
<18:56 NATIVE>
<18:63 FOREACH>
<18:71 INDEX>
<19:1 ASSERT>
<19:8 EMBED>
<19:14 ALIAS>
<19:20 IS>
<19:23 BEGIN>
<19:29 MAIN>
<19:34 HEXBYTES>
<19:43 INC>
<19:47 DEC>
<19:51 UNDERSCORE>
<19:53 OTHERS>
<19:60 WITH>
<19:65 CHECK>
<19:71 GIVES>
<20:1 NOWHERE>
<20:9 INTDIV>
<20:16 LABEL>
<20:22 CLASS>
<20:28 TRAP>
<20:33 EXTENSION>
<20:43 INTERFACE>
<20:53 IMPLEMENTS>
<20:64 UNUSED>
<20:71 ISA>
<21:1 OPTIONAL>
<21:10 IMPORTED>
<21:19 TESTCASE>
<21:28 EXPECT>
<24:1 NUMBER:13>
<25:1 NUMBER:493>
<26:1 NUMBER:48879>
<27:1 NUMBER:1234>
<28:1 NUMBER:1234.5678>
<29:1 NUMBER:123456780>
<30:1 NUMBER:123456780>
<31:1 NUMBER:0.012345678>
<32:1 NUMBER:1234567800000000>
<35:1 STRING:abcd>
<36:1 STRING:a"bcd>
<37:1 STRING:a\bcd>
<38:1 STRING:abcd>
<39:1 STRING:abcd>
<40:1 STRING:a
bcd>
<41:1 STRING:abcd>
<42:1 STRING:a	bcd>
<43:1 STRING:a~bcd>
<44:1 STRING:a~bcd>
<45:1 STRING:aa>
<46:1 STRING:a>
<46:4 SUBBEGIN>
<46:5 IDENTIFIER:x>
<46:6 SUBEND>
<46:7 STRING:bcd>
<47:1 STRING:a>
<47:4 SUBBEGIN>
<47:5 IDENTIFIER:x>
<47:6 SUBFMT>
<47:7 STRING:x>
<47:8 SUBEND>
<47:9 STRING:bcd>
<50:1 STRING:abcd>
<51:1 STRING:abcd>
<52:1 STRING:abcd>
<59:1 IDENTIFIER:a>
<59:7 IDENTIFIER:b>
<60:1 IDENTIFIER:a>
<61:3 IDENTIFIER:b>
<62:1 END_OF_FILE>
